### PR TITLE
[Enhancement] Enable short key for one column filter

### PR DESCRIPTION
**Pull request by kangkaisen**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

I have tested in real user data set, the performance is good.

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -976,4 +976,10 @@ CONF_Int32(default_mv_resource_group_cpu_limit, "1");
 
 // Max size of key columns size of primary key table, default value is 128 bytes
 CONF_mInt32(primary_key_limit_size, "128");
+
+// If your sort key cardinality is very high,
+// You could enable this config to speed up the point lookup query,
+// otherwise, StarRocks will use zone map for one column filter
+CONF_mBool(enable_short_key_for_one_column_filter, "false");
```
Comment (silverbullet233) [inline be/src/common/config.h:983]: can we put it into session variable so that it can be tuned for each query?

**Review by murphyatwork — APPROVED** [review]

File: be/src/common/config.h (hunk 1 above)
Comment (kangkaisen) [reply be/src/common/config.h:983]: I considered this point, which is too hard for user. So I think we could set config for one cluster if this cluster is for point look up

**Review by ZiheLiu — APPROVED** [review]

**Comment by kangkaisen** [comment]
https://github.com/Mergifyio backport branch-3.1

**Comment by satanson** [comment]
@mergifyio backport branch-3.0

**Pull request merged**